### PR TITLE
Dropdown Position suggested mods

### PR DESCRIPTION
**Pull request by spg**

### CODE REVIEW

**Comment by christiangoudreau** [comment]
@jDramaix @meriouma @jclariviere

**Comment by christiangoudreau** [comment]
lgtm

**Comment by jclariviere** [comment]
lgtm

File: plugin/src/main/java/com/arcbees/chosen/client/ChosenImpl.java [hunk 1]
```diff
@@ -1212,18 +1212,22 @@ private boolean resultsShow() {
...
+        searchField.focus();
+
         return true;
     }
 
     private int calculateDropdownTop() {
         int ddTop;
         DropdownPosition dropdownPosition = options.getDropdownPosition();
-        switch (dropdownPosition.getPosition()) {
+        if (dropdownPosition == null) {
```
Comment (meriouma) [inline plugin/src/main/java/com/arcbees/chosen/client/ChosenImpl.java]: Why not set the default dropdown position in the options?

**Comment by meriouma** [comment]
LGTM

File: plugin/src/main/java/com/arcbees/chosen/client/ChosenImpl.java (hunk 1 above)
Comment (jDramaix) [inline plugin/src/main/java/com/arcbees/chosen/client/ChosenImpl.java]: +1 anyway the default case in the switch block uses BELOW

File: plugin/src/main/java/com/arcbees/chosen/client/ChosenImpl.java [hunk 2]
```diff
@@ -1212,18 +1212,22 @@ private boolean resultsShow() {
...
+            dropdownPosition = DropdownPosition.BELOW;
+        }
+        switch (dropdownPosition) {
             case BELOW:
```
Comment (jDramaix) [inline plugin/src/main/java/com/arcbees/chosen/client/ChosenImpl.java]: you should gather the case BELOW with the default:

```java

    case BELOW:
   default:
       ddTop = positionBelow();
}
```

File: plugin/src/main/java/com/arcbees/chosen/client/ChosenOptions.java [hunk 3]
```diff
@@ -155,11 +166,47 @@ public ChosenOptions setSingleBackstrokeDelete(boolean singleBackstrokeDelete) {
...
+     */
     public ChosenOptions setDropdownPosition(DropdownPosition dropdownPosition) {
         this.dropdownPosition = dropdownPosition;
         return this;
     }
 
+    /**
+     * When {@link com.arcbees.chosen.client.ChosenOptions#dropdownPosition} is set to
+     * {@link com.arcbees.chosen.client.DropdownPosition#AUTO}, this element will be used to calculate the available
+     * vertical space below the dropdown.
```
Comment (jDramaix) [inline plugin/src/main/java/com/arcbees/chosen/client/ChosenOptions.java]: you could add that in the javadoic that this option is useless if dropdownposition is not set to AUTO

**Comment by jDramaix** [comment]
2 comments, then LGTM

**Comment by jDramaix** [comment]
I'm merging and I will do the changes myself

**Pull request merged**